src/api/upload.ts

```diff
@@ -0,0 +1,136 @@
+import { getZipFileStream } from './../services/fs';
+import { getElectronFile } from './../services/fs';
+import path from 'path';
+import StreamZip from 'node-stream-zip';
+import { uploadStatusStore } from '../stores/upload.store';
+import { ElectronFile, FILE_PATH_KEYS, FILE_PATH_TYPE } from '../types';
+import { logError } from '../utils/logging';
+import { ipcRenderer } from 'electron';
+import { getSavedFilePaths } from '../utils/upload';
+
+async function getZipEntryAsElectronFile(
+    zip: StreamZip.StreamZipAsync,
+    entry: StreamZip.ZipEntry
+): Promise<ElectronFile> {
+    return {
+        path: entry.name,
+        name: path.basename(entry.name),
+        size: entry.size,
+        lastModified: entry.time,
+        stream: async () => {
+            return await getZipFileStream(zip, entry.name);
+        },
+        blob: async () => {
+            const buffer = await zip.entryData(entry.name);
+            return new Blob([new Uint8Array(buffer)]);
+        },
+        arrayBuffer: async () => {
+            const buffer = await zip.entryData(entry.name);
+            return new Uint8Array(buffer);
+        },
+    };
+}
+
+export const setToUploadFiles = (type: FILE_PATH_TYPE, filePaths: string[]) => {
+    const key = FILE_PATH_KEYS[type];
+    if (filePaths) {
+        uploadStatusStore.set(key, filePaths);
+    } else {
+        uploadStatusStore.delete(key);
```

Comment:
why not have separate function for delete instead of piggy backing on null or empty value?

Reply:
ya should do that will make the change in a new PR